packages/openapi3/src/cli/actions/convert/generators/generate-model.ts

```diff
@@ -1,8 +1,120 @@
-import { TypeSpecModel, TypeSpecModelProperty } from "../interfaces.js";
+import { OpenAPI3Schema, Refable } from "../../../../types.js";
+import {
+  TypeSpecAlias,
+  TypeSpecDataTypes,
+  TypeSpecEnum,
+  TypeSpecModel,
+  TypeSpecModelProperty,
+  TypeSpecScalar,
+  TypeSpecUnion,
+} from "../interfaces.js";
 import { getDecoratorsForSchema } from "../utils/decorators.js";
 import { generateDocs } from "../utils/docs.js";
 import { generateDecorators } from "./generate-decorators.js";
-import { generateTypeFromSchema } from "./generate-types.js";
+import {
+  generateTypeFromSchema,
+  getRefScopeAndName,
+  getTypeSpecPrimitiveFromSchema,
+} from "./generate-types.js";
+
+export function generateDataType(type: TypeSpecDataTypes): string {
+  switch (type.kind) {
+    case "alias":
+      return generateAlias(type);
```

Comment:
when would you get here?

Reply:
Yeah - I didn't realize that Open API 3.0 allows `#/components/schemas` to point to references. Open API 3.1 does not.